src/MergeContext.js

```diff
@@ -499,58 +557,90 @@ class MergeContext {
-                    const matched = requiredChecks.find(el => el.context.startsWith(requiredContext.trim()));
-                    // Before a 'staged' commit can be applied, it should have all required checks passed.
-                    // We create a new "required" check, taking other attributes like targetUrl
-                    // from an already succeeded (matching) check (there can be several matching checks,
-                    // e.g., from different Jenkins nodes). After that, there will be two checks
-                    // referencing the same targetUrl.
-                    if (matched && !this._dryRun("required status check creation"))
-                        await GH.createStatus(ref, "success", matched.targetUrl, matched.description, requiredContext);
-                }
-            }
+
+        return requiredContexts;
+    }
+
+    // returns filled StatusResult object
+    async _getPRStatuses() {
+        let requiredContexts = await this._getRequiredContexts();
+        requiredContexts = requiredContexts.filter(el => el.trim() !== Config.approvalContext());
+        const combinedPrStatus = await GH.getStatuses(this._prHeadSha());
+        let statusResult = new StatusResult(requiredContexts.length, "PR");
+        // fill with required status checks
+        for (let st of combinedPrStatus.statuses) {
+            if (requiredContexts.some(el => el.trim() === st.context.trim() && el.trim() !== Config.approvalContext()))
+                statusResult.addRequiredStatus(new StatusCheck(st));
+        }
+        return statusResult;
+    }
+
+    async _checkPRStatuses(logContext) {
+        const prStatus = await this._getPRStatuses();
+        this._log("status details: " + prStatus);
+        if (prStatus.succeeded())
+            return StepResult.Succeed();
+        // non-final or failed
+        this._log(logContext + " 'status' failed");
+        return StepResult.Fail();
+    }
+
+    // returns filled StatusResult object
+    async _getStagingStatuses() {
+        const combinedStagingStatuses = await GH.getStatuses(this._tagSha);
+        const genuineStatuses = combinedStagingStatuses.statuses.filter(st => !st.description.endsWith(Config.copiedDescriptionSuffix()));
+        assert(genuineStatuses.length <= Config.stagingChecks());
+        let statusResult = new StatusResult(Config.stagingChecks(), "Staging");
+        // all checks are 'required'
+        for (let st of genuineStatuses)
+            statusResult.addRequiredStatus(new StatusCheck(st));
+        return statusResult;
+    }
+
+    async _processStagingStatuses() {
+        const stagingStatus = await this._getStagingStatuses();
+        this._log("status details: " + stagingStatus);
+        if (stagingStatus.failed()) {
+            this._log("staging checks failed");
+            return await this._cleanupMergeFailed(false, this._labelFailedStagingChecks);
+        } else if (!stagingStatus.final()) {
```

Comment:
Please remove the `else` keyword to make this code simpler and more similar to what happens after this/second `if`.